common/functions/src/scalars/udfs/in_basic.rs

```diff
@@ -0,0 +1,258 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashSet;
+use std::fmt;
+
+use common_datavalues::columns::DataColumn;
+use common_datavalues::prelude::DataColumnsWithField;
+use common_datavalues::prelude::MutableArrayBuilder;
+use common_datavalues::prelude::MutableBooleanArrayBuilder;
+use common_datavalues::DataType;
+use common_datavalues::DataTypeAndNullable;
+use common_datavalues::DataValue;
+use common_exception::ErrorCode;
+use common_exception::Result;
+
+use crate::scalars::function_factory::FunctionDescription;
+use crate::scalars::function_factory::FunctionFeatures;
+use crate::scalars::Function;
+
+#[derive(Clone)]
+pub struct InFunction<const NEGATED: bool>;
+
+impl<const NEGATED: bool> InFunction<NEGATED> {
+    pub fn try_create(_display_name: &str) -> Result<Box<dyn Function>> {
+        Ok(Box::new(InFunction::<NEGATED> {}))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        FunctionDescription::creator(Box::new(Self::try_create)).features(
+            FunctionFeatures::default()
+                .bool_function()
+                .variadic_arguments(2, usize::MAX),
```

Comment:
Maybe we can set `variadic_arguments` to 2? The list as an element.

Reply:
sorry, I did not get your point, what do you mean by using struct?

Reply:
For example, `a in (1,2,3)`, `(1,2,3)` can be treated as `DataType::Struct` or `DataType::List`.

Reply:
Get it. 
Hereby using more columns here because we can simply reuse `Expression::ScalarFunction` to simplify parser work here... : ), so the parser will work just as simple

Reply:
The `vec<Expr>` can be parsed to `Tuple` function.